site/content/docs/02-template-syntax.md

```diff
@@ -1132,7 +1132,7 @@ DOMRect {
 
 ---
 
-An animation is triggered when the contents of a [keyed each block](docs#each) are re-ordered. Animations do not run when an element is removed, only when the each block's data is reordered. Animate directives must be on an element that is an *immediate* child of a keyed each block.
+An animation is triggered when an element inside a [keyed each block](docs#each) changes its position. Animations do not run for newly added or removed elements. They only run when the data of an existing element changes position within the each block's data. Animate directives must be on an element that is an *immediate* child of a keyed each block.
```

Comment:
I have to say I'm not a big fan of this change. The passive voice sounds odd, it's more verbose, and it doesn't mean anything different to me

Reply:
Maybe (versus the original one) "Animations do not run when an element is __added or__ removed" (adding the bit about "added or")? I agree that the suggested changes here don't bring much that's new.

Reply:
> it doesn't mean anything different to me
> I agree that the suggested changes here don't bring much that's new.

My main point is to remove the "are re-ordered" part. Because that threw me off and I assumed `animate` only runs when the order of the elements changes. Maybe this is a language barrier, but to me "order" (related to the term "sort order") is defined by the comparator function or some intrinsic property of the list. If I remove an element from the list then nothing is re-ordered. The order does not change.

But like I said, I'm not a native speaker and the only time I come across the term "order" is in a programming/cs context (but I'm aware that there are other orders, such as in a military context or when I order something online, that are entirely unrelated)

The docs say:

> Animations do not run when an element is removed

But _they_ do. The current wording uses the plural form "Animation**s**" when in reality each animation has to be looked at in isolation (because the logic only looks if the index of a given element has changed, ignoring anything around it). Because yes, when an element is removed the animation**s** (plural) _do_ run, but not the animation (singular) of the removed element. Only those animations of elements that come after run.

I assume it is worded this way because the original author knew about transitions and maybe it is worded this way in comparison to transitions (which run when elements are created/removed). But without the transition knowledge the animation paragraph is misleading.

Reply:
How about something like this?

> Animations do not run when an element is added or removed, only when the index of a data item within the each block is changed